### PR TITLE
21w38a terrain generation mappings

### PR DESCRIPTION
**Pull request by mschae23**
Started porting the terrain generation mappings from my Yarn PR.

This also has mappings for some world generation changes in 21w38a.

### CODE REVIEW

**Review by LambdAurora — COMMENTED** [review]
There's 2 empty mapping classes, might want to remove them :P

**Comment by mschae23** [comment]
> There's 2 empty mapping classes, might want to remove them :P

Done.

File: mappings/net/minecraft/world/biome/source/util/OverworldTerrainParameters.mapping [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+		ARG 1 point
+	METHOD m_sioqhtuo getOffsetSpline ()Lnet/minecraft/unmapped/C_ozjstnjc;
+	METHOD m_uqzxhvcl createNoisePoint (FFF)Lnet/minecraft/unmapped/C_mpiahptv$C_wqtllifw;
+		ARG 1 continentalness
+		ARG 2 erosion
+		ARG 3 weirdness
+	METHOD m_yuriezff getFactor (Lnet/minecraft/unmapped/C_mpiahptv$C_wqtllifw;)F
+		ARG 1 point
+	METHOD m_yvpclznw getFactorSpline ()Lnet/minecraft/unmapped/C_ozjstnjc;
+	CLASS C_wqtllifw NoisePoint
```
Comment (OroArmor) [inline mappings/net/minecraft/world/biome/source/util/OverworldTerrainParameters.mapping:23]: is this a record? if so you should match the record naming conventions

Comment (OroArmor) [reply mappings/net/minecraft/world/biome/source/util/OverworldTerrainParameters.mapping:23]: looks like you have it in other places, just want to double check

File: mappings/net/minecraft/util/math/noise/DoublePerlinNoiseSampler.mapping [hunk 2]
```diff
@@ -21,3 +21,15 @@ CLASS net/minecraft/unmapped/C_trjftrxr net/minecraft/util/math/noise/DoublePerl
...
+		FIELD f_kwaudius CODEC Lcom/mojang/serialization/Codec;
+		FIELD f_qmokuuih amplitudes Lit/unimi/dsi/fastutil/doubles/DoubleList;
+		METHOD <init> (ILjava/util/List;)V
+			ARG 1 firstOctave
+			ARG 2 amplitudes
+		METHOD <init> (I[D)V
+			ARG 1 firstOctave
+			ARG 2 amplitudes
+		METHOD m_khfeqjsd amplitudes ()Lit/unimi/dsi/fastutil/doubles/DoubleList;
+		METHOD m_pheowwsb firstOctave ()I
```
Comment (IotaBread) [inline mappings/net/minecraft/util/math/noise/DoublePerlinNoiseSampler.mapping:35]:
```suggestion
		METHOD m_khfeqjsd getAmplitudes ()Lit/unimi/dsi/fastutil/doubles/DoubleList;
		METHOD m_pheowwsb getFirstOctave ()I
```

File: mappings/net/minecraft/world/biome/source/util/MultiNoiseUtil.mapping [hunk 3]
```diff
@@ -1,5 +1,7 @@
 CLASS net/minecraft/unmapped/C_ygdawqho net/minecraft/world/biome/source/util/MultiNoiseUtil
 	FIELD f_akalacgv HYPERCUBE_DIMENSION I
+	METHOD m_aaxqcvjc getParameterAsLong (F)J
```
Comment (IotaBread) [inline mappings/net/minecraft/world/biome/source/util/MultiNoiseUtil.mapping:3]: This one and `getParamterAsFloat` don't seem right

File: mappings/net/minecraft/world/gen/chunk/ChunkNoiseSampler.mapping [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+			ARG 2 noiseZ
+		METHOD m_kcbykdlu sampleNoiseX (D)V
+			ARG 1 deltaX
+		METHOD m_lwxaztkx createBuffer (II)[[D
+			ARG 1 noiseY
+			ARG 2 noiseZ
+		METHOD m_wjebstow sampleNoise ([[DI)V
+			ARG 1 buffer
+			ARG 2 noiseX
+		METHOD m_wpfrryvm sampleNoiseX1 (I)V
```
Comment (IotaBread) [inline mappings/net/minecraft/world/gen/chunk/ChunkNoiseSampler.mapping:137]: These don't seem right?

Comment (mschae23) [reply mappings/net/minecraft/world/gen/chunk/ChunkNoiseSampler.mapping:137]: What do you mean exactly? Also, these aren't new mappings, they're also from Yarn.
Just changed `sampleStartNoise` to `sampleNoiseX0` and `sampleEndNoise` to `sampleNoiseX1`.

**Review by NoComment1105 — APPROVED** [review]
All good, only comments i have just echo what others already said 👍🏼

File: mappings/net/minecraft/util/math/noise/DoublePerlinNoiseSampler.mapping (hunk 2 above)
Comment (OroArmor) [reply mappings/net/minecraft/util/math/noise/DoublePerlinNoiseSampler.mapping:35]: the `Parameter` class might be a record

**Comment by OroArmor** [comment]
I think this PR needs to be remade. The hashed names have 100% changed since this was made, and redoing the port is probably easiest.

**Pull request closed without merging**